### PR TITLE
Add sort functionality

### PR DESCRIPTION
**Pull request by wz27**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 1]
```diff
@@ -10,6 +10,7 @@
     public static final Prefix PREFIX_PHONE = new Prefix("p/");
     public static final Prefix PREFIX_EMAIL = new Prefix("e/");
     public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
+    public static final Prefix PREFIX_SORT = new Prefix("-r");
```
Comment (LeopardMerkava) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:13]: Is there a use for this? Is it a prefix used parser?

**Review by LeopardMerkava — COMMENTED** [review]
There are a few things I pointed out. Mostly minor things. Don't forget to add the tests later!

File: src/main/java/seedu/address/model/person/Person.java [hunk 2]
```diff
@@ -120,4 +120,9 @@ public String toString() {
         return builder.toString();
     }
 
+    @Override
+    public int compareTo(Object other) {
+        Person o = (Person) other;
```
[the remark this answers is not in the document]
Comment (wz27) [reply src/main/java/seedu/address/model/person/Person.java]: Thanks for pointing it out. Updated.

File: src/main/java/seedu/address/logic/parser/CliSyntax.java (hunk 1 above)
Comment (wz27) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java]: Thanks for pointing it out. Updated.

**Comment by wz27** [comment]
Sort and reverse sort functionality have been implemented. Tests are not yet completed.

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sorts the list of persons "
+            + "by the alphabetical order of their name.\n"
+            + "Parameters: [-r]\n"
+            + "Example: " + COMMAND_WORD
+            + "-r";
+
+    private String message_success = "Sorted all persons";
+
+    private final boolean reverseOrder;
```
Comment (limzk126) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:22]: I think changing this boolean variable name to isReverseOrder would improve code readability.

File: src/main/java/seedu/address/logic/parser/SortCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * Parses input arguments and creates a new FindCommand object
+ */
+public class SortCommandParser implements Parser<SortCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SortCommand parse(String args) throws ParseException {
```
Comment (eugenecsa) [inline src/main/java/seedu/address/logic/parser/SortCommandParser.java:16]: Is ParseException supposed to be thrown by the below code?

**Review by eugenecsa — COMMENTED** [review]
The code looks fine, just like to clarify about the part with the error as i am unsure where the error is thrown from.

**Review by limzk126 — COMMENTED** [review]
Overall LGTM 👍🏻

File: src/main/java/seedu/address/model/ModelManager.java [hunk 5]
```diff
@@ -21,7 +23,8 @@
 
     private final AddressBook addressBook;
     private final UserPrefs userPrefs;
-    private final FilteredList<Person> filteredPersons;
+    private final FilteredList<Person> onlyfilteredPersons;
+    private final SortedList<Person> filteredPersons;
```
[the remark this answers is not in the document]
Comment (wz27) [reply src/main/java/seedu/address/model/ModelManager.java:27]: Thanks for pointing it out! I avoided renaming filteredPersons in this iteration as other enhancements may have naming dependencies on it.

File: src/main/java/seedu/address/logic/parser/SortCommandParser.java (hunk 4 above)
Comment (wz27) [reply src/main/java/seedu/address/logic/parser/SortCommandParser.java:16]: Thanks for pointing it out! Updated.

**Comment by kflim** [comment]
LGTM, nothing major of concern

**Pull request merged**